abigen/src/lib.rs

```diff
@@ -245,11 +245,12 @@ fn to_token(name: &proc_macro2::TokenStream, kind: &ParamType) -> proc_macro2::T
         }
         ParamType::Bytes => quote! { ethabi::Token::Bytes(#name.clone()) },
         ParamType::FixedBytes(_) => quote! { ethabi::Token::FixedBytes(#name.as_ref().to_vec()) },
-        ParamType::Int(_) => {
+        ParamType::Int(size) => {
+            let full_signed_bytes_init = if size == 128 { 0x00 } else { 0xff };
```

Comment:
The `128` check feels weird, why a `i128` would be different than a `i64`. I'm surprised here.

If you use `0xff` directly.